etl/chart_revision/v2/updaters/variable_update.py

```diff
@@ -0,0 +1,471 @@
+        List[gm.Chart]
+            List of charts that use the variables that are being updated.
+        """
+        # Get charts to be updated
+        charts = find_charts_from_variable_ids(self.variable_ids_old)
+        # Get metadata for variables in use in the charts
+        self.__variable_meta = self._get_variable_metadata(charts)
+        return charts
+
+    def _get_variable_metadata(self, charts: List[gm.Chart]) -> Dict[int, Any]:
+        """Get variable metadata.
+
+        We need to get metadata for variables being updated (old and new), but also for variables that are not being updated, but are used in the charts.
+        This is because we need to know the min and max years for all variables used in the charts to update the chart time configuration.
+        """
+        # Get variable IDs from all variables (dimensions x and y) in charts that will be updated
+        variable_ids = set()
+        for chart in charts:
+            variable_ids |= set(c["variableId"] for c in chart.config["dimensions"] if c["property"] in {"x", "y"})
+        log.info(
+            f"variable_update: getting variable metadata of {len(variable_ids)} variables from {len(charts)} charts"
+        )
+        # Combine with new variables (not yet used in charts)
+        variable_ids = list(variable_ids | set(self.variable_ids_new))
+
+        # Get metadata of variables from S3k
+        df = variable_data_df_from_s3(get_engine(), variable_ids=[int(v) for v in variable_ids], workers=10)
+        # Reshape metadata, we want a dictionary!
+        variable_meta = (
+            df.groupby("variableId").year.agg(["min", "max"]).rename(columns={"min": "minYear", "max": "maxYear"})
+        )
+        variable_meta = variable_meta.to_dict(orient="index")
+        return variable_meta
+
+    def run(self, config: Dict[str, Any]) -> Dict[str, Any]:
+        """Run the chart variable updater."""
+        log.info("variable_update: updating configuration")
+        # Validate the configuration of the chart and add default values
+        config_new = validate_chart_config_and_set_defaults(config, self.schema)
+        # Update map configuration
+        config_new = update_config_map(
+            config_new,
+            self.variable_mapping,
+            self.variable_meta,
+            config_new["dimensions"][0]["variableId"],
+        )
+        # Update config time
+        config_new = update_config_time(config_new, self.variable_mapping, self.variable_meta)
+        # Update dimensions
+        config_new = update_config_dimensions(config_new, self.variable_mapping)
+        # Update sort
+        config_new = update_config_sort(config_new, self.variable_mapping)
+        # Validate the  configuration of the chart and remove default values (if any)
+        config_new = validate_chart_config_and_remove_defaults(config_new, self.schema)
+        return config_new
+
+
+def find_charts_from_variable_ids(variable_ids: Set[int]) -> List[gm.Chart]:
```

Comment:
This could go into grapher_helpers as a classmethod of `Variable`, might be useful elsewhere.

Reply:
Interesting! 

Did you mean `grapher_model`? I can't find `Variable` class in `grapher_helpers`.

If we were to put it under `Variable` in `grapher_model`, it would only be able to retrieve charts used by a single variable with one DB call, right?

As it is now, with one DB call we query for multiple variables.

Maybe we can just place this function in `grapher_helpers` as it is now. ? @Marigold

Reply:
Ugh, sorry, I meant `grapher_model.py`. There's an example of this method
```method
@classmethod
    def load_variable(cls, session: Session, variable_id: int) -> "Variable":
        return session.exec(select(cls).where(cls.id == variable_id)).one()
```
which is useful syntactic sugar for getting variable by ID. We could have similar method for `find_charts_from_variable_ids`. Having these methods on sqlmodel objects makes them more discoverable and reusable.